### PR TITLE
Release 1.5.0 add new SDK

### PR DESCRIPTION
**Pull request by 2bogd**

### CODE REVIEW

File: Example/SCard/ViewController.swift [hunk 1]
```diff
@@ -44,20 +44,23 @@ class ViewController: UIViewController {
 
         refreshBalanceTimer.invalidate()
         refreshBalanceTimer = Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { _ in
-            let balane = Decimal(UInt.random(in: 2500...5000))
+            let balane = Decimal(UInt.random(in: 250000...500000))
```
Comment (Viscount88) [inline Example/SCard/ViewController.swift]: balance*

File: Example/SCard/ViewController.swift [hunk 2]
```diff
@@ -44,20 +44,23 @@ class ViewController: UIViewController {
...
             pwAuthDomain: "soracard.com",
             pwApiKey: "",
+            appPlatformId: "",
+            recaptchaKey: "",
             kycUrl: "https://kyc-test.soracard.com/mobile",
             kycUsername: "",
             kycPassword: "",
             xOneEndpoint: "https://dev.x1ex.com/widgets/sdk.js",
-            xOneId: "",
+            xOneId: "sprkwdgt-WYL6QBNC",
```
Comment (Viscount88) [inline Example/SCard/ViewController.swift:65]: Is it okay that it's open?

File: SCard/Classes/Modules/SCApiClient/SCAPIClient.swift [hunk 3]
```diff
@@ -116,7 +117,17 @@ public class SCAPIClient {
...
+            PayWingsOAuthClient.instance()?.getNewAuthorizationData(
+                methodUrl: "/test", httpRequestMethod: .POST, completion: { authData in
+                continuation.resume(returning: (authData.accessTokenData?.accessToken, authData.dpop))
+                if authData.userSignInRequired ?? false {
+                    print("SCAPIClient userSignInRequired")
+                }
+            })
+        }
+
+        urlRequest.addValue("Bearer " + (accessToken ?? ""), forHTTPHeaderField: "Authorization")
```
Comment (Viscount88) [inline SCard/Classes/Modules/SCApiClient/SCAPIClient.swift]: Does it work with empty token? Probably we should send it only if we got our token.

File: SCard/Classes/Modules/SCKYCEnterEmailCode/SCKYCEnterEmailCodeViewModel.swift [hunk 4]
```diff
@@ -44,29 +44,27 @@ final class SCKYCEnterEmailCodeViewModel {
...
-            await MainActor.run {
-                self.onContinue?(self.data)
-            }
-        }
+        onContinue?(self.data)
     }
 
-    func onUserSignInRequired() {
+    func onEmailNotVerified() {
+        print("SCKYCEnterEmailCodeViewModel onEmailNotVerified")
```
Comment (Viscount88) [inline SCard/Classes/Modules/SCKYCEnterEmailCode/SCKYCEnterEmailCodeViewModel.swift:53]: Replace print to log

File: SCard/Classes/Modules/SCKYCEnterEmailCode/SCKYCEnterEmailCodeViewModel.swift [hunk 5]
```diff
@@ -44,29 +44,27 @@ final class SCKYCEnterEmailCodeViewModel {
...
     }
+}
+
+extension SCKYCEnterEmailCodeViewModel: SendNewVerificationEmailCallbackDelegate {
 
     func onError(error: PayWingsOAuthSDK.OAuthErrorCode, errorMessage: String?) {
+        print("SCKYCEnterEmailCodeViewModel error:\(error)")
```
Comment (Viscount88) [inline SCard/Classes/Modules/SCKYCEnterEmailCode/SCKYCEnterEmailCodeViewModel.swift:60]: Replace print to log

File: SCard/Classes/Modules/SCKYCEnterPhone/SCKYCEnterPhoneViewModel.swift [hunk 6]
```diff
@@ -78,6 +83,10 @@ final class SCKYCEnterPhoneViewModel {
 }
 
 extension SCKYCEnterPhoneViewModel: SignInWithPhoneNumberRequestOtpCallbackDelegate {
+    func onShowTimeBasedOtpVerificationInputScreen(accountName: String) {
+        print("TODO: onShowTimeBasedOtpVerificationInputScreen")
```
Comment (Viscount88) [inline SCard/Classes/Modules/SCKYCEnterPhone/SCKYCEnterPhoneViewModel.swift:87]: Replace print to log

File: SCard/Classes/Modules/SCKYCEnterPhoneCode/SCKYCEnterPhoneCodeViewModel.swift [hunk 7]
```diff
@@ -54,6 +54,24 @@ final class SCKYCEnterPhoneCodeViewModel {
...
+            guard let self = self else { return }
+            self.service.getUserData(callback: self.getUserDataCallback)
+            await MainActor.run {
+                self.onUpdateUI?()
+            }
+        }
+    }
+    
+    func onShowTimeBasedOtpVerificationInputScreen(accountName: String) {
+        print("TODO: onShowTimeBasedOtpVerificationInputScreen")
```
Comment (Viscount88) [inline SCard/Classes/Modules/SCKYCEnterPhoneCode/SCKYCEnterPhoneCodeViewModel.swift:68]: Replace print to log

File: SCard/Classes/Modules/SCKYCOnbording/SCKYCOnbordingViewModel.swift [hunk 8]
```diff
@@ -17,7 +20,25 @@ final class SCKYCOnbordingViewModel {
...
 
-    func fetchReferenceNumber() async -> Bool {
+    func set(kycId: String) {
+        data.kycId = kycId
+        storage.add(kycId: kycId)
+    }
+
+    func startKYC() {
+        Task {
+            guard await checkCameraPermission() else { return }
```
Comment (Viscount88) [inline SCard/Classes/Modules/SCKYCOnbording/SCKYCOnbordingViewModel.swift:31]: Probably we should show some alert if unavailable and suggest to allow

File: SCard/Classes/Modules/SCKYCOnbording/SCKYCOnbordingViewModel.swift [hunk 9]
```diff
@@ -44,107 +70,87 @@ final class SCKYCOnbordingViewModel {
...
-
-            let credentials = KycCredentials(
-                username: service.config.kycUsername,
-                password: service.config.kycPassword,
-                endpointUrl: service.config.kycUrl
-            )
+        let cameraAuthorized = (AVCaptureDevice.authorizationStatus(for: AVMediaType.video) == .authorized) ? true : false
+        let microphoneAuthorized = (AVCaptureDevice.authorizationStatus(for: AVMediaType.audio) == .authorized) ? true : false
+
+        guard let viewController = viewController, cameraAuthorized, microphoneAuthorized else {
```
Comment (Viscount88) [inline SCard/Classes/Modules/SCKYCOnbording/SCKYCOnbordingViewModel.swift]: Would better to separate these errors in future

File: SCard/Classes/Modules/SCKYCOnbording/SCKYCOnbordingViewModel.swift (hunk 8 above)
Comment (2bogd) [reply SCard/Classes/Modules/SCKYCOnbording/SCKYCOnbordingViewModel.swift:31]: encapsulated inside function

File: SCard/Classes/Modules/SCKYCEnterPhoneCode/SCKYCEnterPhoneCodeViewModel.swift (hunk 7 above)
Comment (2bogd) [reply SCard/Classes/Modules/SCKYCEnterPhoneCode/SCKYCEnterPhoneCodeViewModel.swift:68]: did add global proxy func

**Review by Viscount88 — APPROVED** [review]

**Pull request merged**